nixos/modules/services/games/vrising-server.nix

```diff
@@ -0,0 +1,176 @@
+              '';
+            };
+          };
+        };
+        default = { };
+        example = literalExpression ''
+          {
+            Name = "V Rising Server";
+            Description = "";
+            MaxConnectedUsers = 5;
+            MaxConnectedAdmins = 5;
+            ServerFps = 30;
+            SaveName = "world1";
+            Secure = true;
+            ListOnMasterServer = false;
+            AutoSaveCount = 20;
+            AutoSaveInterval = 300;
+            GameSettingsPreset = "StandardPvE";
+            AdminOnlyDebugEvents = true;
+            DisableDebugEvents = false;
+          }'';
+        description = ''
+          Host settings as defined in <link xlink:href="https://github.com/StunlockStudios/vrising-dedicated-server-instructions">this repository</link>.
+          Look at the reference to see what you can configure in this attribute set.
+
+          Do not set <literal>Password</literal> here but use the dedicated <literal>passwordFile</literal> option instead.
+        '';
+      };
+
+      gameSettings = mkOption {
+        type = format.type;
+        default = { };
+        example = literalExpression ''
+          {
+            GameModeType = "PvE";
+            DeathContainerPermission = "ClanMembers";
+            RelicSpawnType = "Plentiful";
+            CastleHeartDamageMode = "CanBeDestroyedOnlyWhenDecaying";
+            CastleDamageMode = "Never";
+            BloodBoundEquipment = true;
+            CanLootEnemyContainers = false;
+            Death_DurabilityFactorLoss = 0.0;
+            Death_DurabilityLossFactorAsResources = 0.0;
+            AnnounceSiegeWeaponSpawn = false;
+          }'';
+        description = ''
+          Game settings as defined in <link xlink:href="https://cdn.stunlock.com/blog/2022/05/25083113/Game-Server-Settings.pdf">this file</link>.
+          Look at the reference to see what you can configure in this attribute set.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    warnings = optional (cfg.hostSettings ? Password) ''
+      You have defined a host password using the hostSettings option, which
+      will be ignored. Use the dedicated passwordFile option instead.
+    '';
+
+    networking.firewall.allowedUDPPorts = mkIf cfg.openFirewall [
+      cfg.hostSettings.port
+      cfg.hostSettings.queryPort
+    ];
+
+    systemd.services.vrising-server = {
+      description = "V Rising dedicated server";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network-online.target" ];
+
+      environment = {
+        # steamcmd uses $HOME/.local for temporary files
+        HOME = varLibStateDir;
+        saveDir = "./save-data";
+      };
+
+      serviceConfig = {
+        ExecStartPre = ''
+          ${pkgs.steamcmd}/bin/steamcmd \
+          +force_install_dir ${varLibStateDir} \
+          +login anonymous \
+          +app_update 1829350 \
+          +quit
```

Comment:
That is really ugly and really statefull. Don't they have another download method?

Reply:
Unfortunately no, at least none that I know of.